packages/in_app_purchase/in_app_purchase_android/lib/src/billing_client_wrappers/pending_purchases_params_wrapper.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+// WARNING: Changes to `@JsonSerializable` classes need to be reflected in the
+// below generated file. Run `flutter packages pub run build_runner watch` to
+// rebuild and watch for further changes.
+part 'pending_purchases_params_wrapper.g.dart';
+
+/// Dart wrapper around [`com.android.billingclient.api.PendingPurchasesParams`](https://developer.android.com/reference/com/android/billingclient/api/PendingPurchasesParams).
+///
+/// Represents the parameters to enable pending purchases.
+@JsonSerializable()
+@immutable
+class PendingPurchasesParamsWrapper {
+  /// Creates a [PendingPurchasesParamsWrapper].
+  const PendingPurchasesParamsWrapper({
+    required this.enablePrepaidPlans,
+  });
+
+  /// Factory for creating a [PendingPurchasesParamsWrapper] from a [Map].
+  @Deprecated('JSON serialization is not intended for public use, and will '
```

Comment:
Why are we adding new code that is pre-deprecated?

The reason other classes had JSON serialization is that they were being serialized as JSON to send over the method channel. The reasons it's deprecated in all those classes is that we don't do that any more, because we use Pigeon. It was only left in place was to avoid a breaking change on the off-chance someone was using it for other reasons, since it was part of the public interface.